analyzer/src/main/kotlin/managers/PubCacheReader.kt

```diff
@@ -82,20 +82,18 @@ internal class PubCacheReader {
             log.error { "Could not find projectRoot of '$packageName'." }
 
             // Unsupported type.
-            null
+            return null
         }
 
-        if (path != null) {
-            pubCacheRoot.resolve(path).let {
-                if (it.isDirectory) {
-                    return it
-                }
+        pubCacheRoot.resolve(path).let {
+            if (it.isDirectory) {
+                return it
             }
+        }
```

Comment:
How about instead (using `also` instead of `let` as the return value from the lambda is irrelevant):

    pubCacheRoot.resolve(path).takeIf { it.isDirectory }?.also { return it }

Similar below.

Reply:
I've made a separate commit for this on top based on your idea.